tensorboard/webapp/plugins/npmi/views/data_selection/metric_search/metric_search_container.ts

```diff
@@ -0,0 +1,111 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+import {Component, ChangeDetectionStrategy} from '@angular/core';
+import {select, Store} from '@ngrx/store';
+
+import {map, combineLatest} from 'rxjs/operators';
+
+import {State} from '../../../../../app_state';
+import {
+  getMetricFilterKeys,
+  getMetricsRegex,
+  getRunToMetrics,
+} from '../../../store';
+import {getRunSelection} from '../../../../../core/store/core_selectors';
+import * as npmiActions from '../../../actions';
+
+/** @typehack */ import * as _typeHackRxjs from 'rxjs';
+
+@Component({
+  selector: 'npmi-metric-search',
+  template: `
+    <metric-search-component
+      [regexFilterValue]="metricsRegex$ | async"
+      [completions]="completions$ | async"
+      [isRegexFilterValid]="isMetricsFilterValid$ | async"
+      (onRegexFilterValueChange)="onFilterChange($event)"
+    ></metric-search-component>
+  `,
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class MetricSearchContainer {
+  readonly metricsRegex$ = this.store.select(getMetricsRegex);
+  readonly activeRuns$ = this.store.pipe(select(getRunSelection)).pipe(
+    map((runs) => {
+      let activeRuns: string[] = [];
+      if (runs) {
+        for (let run of runs) {
+          if (run[1]) {
+            activeRuns.push(run[0]);
+          }
+        }
+      }
+      return activeRuns;
+    })
+  );
+  readonly getRunToMetrics$ = this.store.pipe(select(getRunToMetrics));
+  readonly completions$ = this.store
+    .pipe(select(getRunToMetrics))
+    .pipe(
+      combineLatest(this.activeRuns$),
+      map(([runToMetrics, activeRuns]) => {
+        const metrics: string[] = [];
+        for (const run of activeRuns) {
+          if (runToMetrics[run]) {
+            for (const metric of runToMetrics[run]) {
+              if (!metrics.includes(metric)) {
+                metrics.push(metric);
+              }
+            }
+          }
+        }
+        return metrics;
+      })
+    )
+    .pipe(
+      combineLatest(this.store.pipe(select(getMetricFilterKeys))),
+      map(([metrics, metricsActive]) => {
+        return metrics.filter(
+          (metric: string) => !metricsActive.includes(metric)
+        );
+      })
+    )
+    .pipe(
+      combineLatest(this.metricsRegex$),
+      map(([metrics, regexFilterValue]) => {
+        try {
+          const filterRegex = new RegExp(regexFilterValue, 'i');
+          return metrics
+            .filter((metric: string) => filterRegex.test(metric))
+            .sort();
+        } catch (err) {
+          return metrics.sort();
+        }
+      })
+    );
+  readonly isMetricsFilterValid$ = this.metricsRegex$.pipe(
+    map((filterString) => {
+      try {
+        // tslint:disable-next-line:no-unused-expression Check for validity of filter.
```

Comment:
Alternatively, can we have the try block do something like:
`return Boolean(new RegExp(filterString))`

Reply:
Is that safe, though? Can that cast never be false even if the regex exists?

Reply:
Great, changed it.